### PR TITLE
refactor(*) create common function for fetching data

### PR DESCRIPTION
**Pull request by tomaszwylezek**
### Summary
It will align the way how the data is retrieved from API to be displayed in the GUI. Right now we have a lot of duplication of code. This is one of many steps to make sure we can better manage the data/duplication of code.

Signed-off-by: Tomasz Wylężek <tomwylezek@gmail.com>

### CODE REVIEW

File: src/utils/tableDataUtils.ts [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+function sortEntities(
+  items: {
+    name: string
+    mesh: string
+  }[],
+) {
+  return [...items].sort((a, b) => (a.name > b.name ? 1 : a.name === b.name ? (a.mesh > b.mesh ? 1 : -1) : -1))
+}
+
+const getItems = (response: {
```
Comment (Pan-Bubr) [inline src/utils/tableDataUtils.ts:13]: Could we add a type signature? I am not sure what are we returning here

File: src/utils/tableDataUtils.ts [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+  total: number
+  items: {
+    name: string
+    mesh: string
+  }[]
+}) => {
+  const r = response
+
+  if (!r.total) {
+    return [r]
```
Comment (Pan-Bubr) [inline src/utils/tableDataUtils.ts:23]: If the total is 0 we return whole response as a single-element Array and lower we return [...items]

File: src/utils/tableDataUtils.ts [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+  query,
+  size,
+  offset,
+}: TableDataParams) {
+  const params = {
+    size,
+    offset,
+  }
+
+  if (mesh === 'all') {
```
Comment (Pan-Bubr) [inline src/utils/tableDataUtils.ts:47]: Do we need an await here? We can just return promise.

**Review by Pan-Bubr — CHANGES_REQUESTED** [review]

File: src/utils/tableDataUtils.ts [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+}
+
+const getItems = (response: {
+  total: number
+  items: {
+    name: string
+    mesh: string
+  }[]
+}) => {
+  const r = response
```
Comment (Pan-Bubr) [inline src/utils/tableDataUtils.ts:20]: What is the purpose of this assignment?

File: src/utils/tableDataUtils.ts (hunk 2 above)
Comment (Pan-Bubr) [reply src/utils/tableDataUtils.ts:23]: Probably
`return [...r.items]`

File: src/utils/tableDataUtils.ts [hunk 5]
```diff
@@ -0,0 +1,88 @@
+import { getOffset } from '@/helpers'
+import { TableDataParams } from './tableDataUtils.types'
+
+function sortEntities(
+  items: {
```
Comment (Pan-Bubr) [inline src/utils/tableDataUtils.ts:5]: Items are used in a lot of places, i suggest that they deserve an Interface

File: src/utils/tableDataUtils.ts (hunk 3 above)
Comment (tomaszwylezek) [reply src/utils/tableDataUtils.ts:47]: Good point, we don't need await there

File: src/utils/tableDataUtils.ts (hunk 4 above)
Comment (tomaszwylezek) [reply src/utils/tableDataUtils.ts:20]: None, just copy-paste of function from which it was taken. Fixed.

**Review by Pan-Bubr — APPROVED** [review]

**Pull request merged**